ch-05/src/Solution58.java

```diff
@@ -0,0 +1,26 @@
+import java.util.Scanner;
+
+public class Solution58 {
+    public static void main(String[] args) {
+        final Scanner input = new Scanner(System.in);
+
+        System.out.println("Enter the number of students:");
+        final int studentsNumber = input.nextInt();
+        String StudentsNameWhoScoredTheHighest = "";
+        int previousStudentsScore = 0;

```

Comment:
The name of this variable is a bit misleading, as it holds the largest score seen so far.